torch_xla/csrc/cross_replica_reduces.cpp

```diff
@@ -259,4 +280,18 @@ ReduceScatterResult BuildReduceScatter(
   return {reduce_result, token_handler.GetNewToken(reduce_result)};
 }
 
+const torch::lazy::Value& GetAllReduceToken(
+    const torch::lazy::BackendDevice& device) {
+  if (!g_all_reduce_token.token) {
+    std::lock_guard<std::mutex> lock(g_all_reduce_token.lock);
+    g_all_reduce_token.token = CreateToken(device);
+  }
+  return *g_all_reduce_token.token;
+}
+
+void SetAllReduceToken(const std::shared_ptr<torch::lazy::Value>& token) {
+  std::lock_guard<std::mutex> lock(g_all_reduce_token.lock);
```

Comment:
Hmm, why do we need lock here? I guess `mark_step` will set this token to None, but there shouldn't be any async process trying to access this token IR right? Are you just trying to be extra safe?

Reply:
Yea, just to be safe. I can remove it if you think it's not necessary...

Reply:
Don't want to introduce any noise here, but we can consider removing it later.

Reply:
Already removed.